tests/unit/test_rose_stem_units.py

```diff
@@ -54,3 +73,225 @@ def test_get_source_opt_from_args(tmp_path, monkeypatch, args, expect):
         assert result == expect
     else:
         assert result == expect.format(tmp_path=str(tmp_path))
+
+
+@pytest.fixture
+def get_StemRunner():
+    def _inner(kwargs, options=None):
+        if options is None:
+            options = {}
+        """Create a StemRunner objects with some options set."""
+        opts = SimpleNamespace(**{'verbosity': 1, 'quietness': 1})
+        for k, v in options.items():
+            setattr(opts, k, v)
+        stemrunner = StemRunner(opts, **kwargs)
+        return stemrunner
+    return _inner
+
+
+@pytest.mark.parametrize(
+    'attribute, object_, kwargs', [
+        param('reporter', Reporter, {}, id='reporter default'),
+        param('reporter', str, {'reporter': 'foo'}, id='reporter set'),
+        param('popen', RosePopener, {}, id='popen default'),
+        param('popen', str, {'popen': 'foo'}, id='popen set'),
+        param('fs_util', FileSystemUtil, {}, id='fs_util default'),
+        param('fs_util', str, {'fs_util': 'foo'}, id='fs_util set')
+    ]
+)
```

Comment:
This creates one StemRunner per test run which isn't necessary.

Ok to ignore but IMO this would be better/clearer without parametrize. If one of the subtests goes wrong it is easier to work out what's broken when the traceback shows something like:

```
assert isinstance(stem_runner.reporter, Reporter)
Traceback ...
AssertionError: NoneType != Reporter
```

Rather than `assert isinstance(item, object_)`, then having to find the parametrised part of the test name that failed, then get into the source code to lookup the values of `item` and `object_` to find out what actually went wrong.

<img src="http://media.gettyimages.com/photos/hammer-hitting-bent-screw-into-wood-picture-id181197695?s=612x612" />

Reply:
Mea Cupa